src/python/pants/backend/native/tasks/conan_prep.py

```diff
@@ -15,3 +16,6 @@ class ConanInstance(PythonToolInstance):
 class ConanPrep(PythonToolPrepBase):
   tool_subsystem_cls = Conan
   tool_instance_cls = ConanInstance
+
+  def will_be_invoked(self):
+    return len(self.get_targets(lambda t: isinstance(t, ExternalNativeLibrary))) > 0
```

Comment:
Probably a bit more idiomatic use of the language: `any(self.get_targets(...))`.

Reply:
Oops, was going a little fast there. Will do.